File: 1000-9999/5615.py
'''
5615번
아파트 임대
'''

import sys
input=sys.stdin.readline

def modPow(a, n, l):
    if n==0: return 1
    res=modPow(a*a%l, n//2, l)
    if n&1: res=res*a%l
    return res

def mr(n, a):
    if a%n==0: return True
    d=n-1
    while True:
        p=modPow(a, d, n)
        if p==n-1: return True
        if d%2: return (p==1 or p==n-1)
        d//=2

def isPrime(n):
    if n<=3: return False

    base=[2, 7, 61]
    for a in base:
        if not mr(n, a): return False
    return True

t=int(input())

ans=0
for _ in range(t):
    n=int(input())
    if isPrime(2*n+1): ans+=1

print(ans)

File: 1000-9999/5615.cpp
#include <bits/stdc++.h>
using namespace std;
#define ull unsigned long long

ull n, a, ans;

ull mod_pow(ull a, ull n, ull l){
    if(n==0) return 1;
    ull res=mod_pow(a*a%l, n/2, l);
    if(n&1) res=((res%l)*(a%l))%l;
    return res;
}

bool mr(ull n, ull a){
    if(a%n==0) return true;
    ull d=n-1;
    while(1){
        ull tmp=mod_pow(a, d, n);
        if(tmp==n-1) return true;
        if(d%2) return (tmp==1 || tmp==n-1);
        d/=2;
    }
}

bool is_prime(ull n){
    ull base[3]={2, 7, 61};
    for(int i=0; i<3; i++){
        if(!mr(n, base[i])) return false;
    }
    return true;
}

int main(void){
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);

    cin>>n;
    while(n--){
        cin>>a;
        if(is_prime(a*2+1)) ans++;
    }
    cout<<ans<<"\n";
}